include/ppx/application.h

```diff
@@ -523,13 +524,28 @@ class Application
     // See StartMetricsRun for why this wrapper is necessary.
     virtual bool HasActiveMetricsRun() const;
 
+    // Allocate a metric id to be used for a combined live/recorded metric.
+    metrics::MetricID AllocateMetricID();
+
     // Adds a metric to the current run. If no run is active, returns metrics::kInvalidMetricID.
     // See StartMetricsRun for why this wrapper is necessary.
-    virtual metrics::MetricID AddMetric(const metrics::MetricMetadata& metadata);
+    metrics::MetricID AddMetric(const metrics::MetricMetadata& metadata);
+
+    // Bind a metric to the current run, return false if no run is active.
+    bool BindMetric(metrics::MetricID metricID, const metrics::MetricMetadata& metadata);
+
+    // Bind a live metric, return true on success.
+    bool BindLiveMetric(metrics::MetricID metricID);
```

Comment:
Could you elaborate here? What does "binding" a metric mean? Why do we need to call that? What happens if we don't? 
I would ideally like to know these things without reading the code.